[FancyZones] Implement File Watcher

ivan100sic opened the pull request:
## Summary of the Pull Request

This PR prepares the terrain for upcoming changes to how the FZ Editor and FancyZones communicate.
Effectively, this PR allows you to change the `zones-settings.json` file in any editor, save it, and have that change be reflected
in FancyZones without restarting it - it will happen within 1 second by default.

## PR Checklist
* [x] Applies to: it's part of the work needed to support the new editor UI https://github.com/microsoft/PowerToys/pull/8148
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/PowerToys) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Info on Pull Request

A new utility class - FileWatcher, and supporting code, mostly in `FancyZones.cpp`

## Validation Steps Performed

Make sure that FZ Editor still works and changing the `zones-settings.json` file from some text editor works.

ivan100sic committed:
Implement File Watcher in FancyZones

yuyoyuppe reviewed (changes requested):
Nice work, but also let's address some comments!

ivan100sic commented on src/modules/fancyzones/lib/FileWatcher.h:
```diff
@@ -0,0 +1,90 @@
+
+#include "pch.h"
+
+class FileWatcher
```
I agree. It used to be a template class.

ivan100sic commented on src/modules/fancyzones/lib/FileWatcher.h:
```diff
@@ -0,0 +1,90 @@
+
+    std::optional<FILETIME> MyFileTime()
+    {
+        HANDLE hFile = CreateFileW(m_path.c_str(), 0, FILE_SHARE_DELETE | FILE_SHARE_READ | FILE_SHARE_WRITE, NULL, OPEN_EXISTING, 0, NULL);
```
Regarding the first one, sure.
Specified file attributes will be ignored when using `OPEN_EXISTING`.

ivan100sic committed:
Simplify code, address PR comments

ivan100sic committed:
Add check to result of CreateEventW

ivan100sic committed:
Rebase fix

Removed unneeded newline. If we keep it now, VS will just remove it some other time.

yuyoyuppe commented:
/azp run
